### PR TITLE
Replaced relative link with absolute links and remove locale

### PR DESCRIPTION
**Pull request by sima-zhu**
The PR is to turn on the link check flag.
1. Disable the relative links.
2. Remove locale "en-us" from link.

Pipeline: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=541050&view=logs&j=ad792ccc-30b2-59a1-5b53-546a79d88828&t=98098dc8-5acb-5f94-5428-670dacfec5e2

### CODE REVIEW

**Review by alzimmermsft — COMMENTED** [review]
Many links are using `http` instead of `https`

**Comment by sima-zhu** [comment]
New pipeline: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=541790&view=results

**Review by mssfang — APPROVED** [review]
LGTM for AppConfig, TextAnalytics and FormRecognizer changes! Thanks Sima.

**Review by alzimmermsft — APPROVED** [review]

**Review by samvaity — APPROVED** [review]

**Review by weidongxu-microsoft — APPROVED** [review]
Thanks for the help!

File: sdk/eventhubs/azure-messaging-eventhubs/README.md [hunk 1]
```diff
@@ -81,7 +81,7 @@ Both the asynchronous and synchronous Event Hub producer and consumer clients ca
 
 The snippet below creates a synchronous Event Hub producer.
 
-<!-- embedme ./src/samples/java/com/azure/messaging/eventhubs/ReadmeSamples.java#L31-L35 -->
+<!-- embedme https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/eventhubs/azure-messaging-eventhubs/src/samples/java/com/azure/messaging/eventhubs/ReadmeSamples.java#L31-L35 -->
```
Comment (weshaggard) [inline sdk/eventhubs/azure-messaging-eventhubs/README.md:84]: @sima-zhu  how did you find these links? I wouldn't have expected them to be flagged by our checks as they aren't real links from an html prospective.

**Comment by weshaggard** [comment]
I was expecting you to enable the link guidance as part of this PR so that we can confirm that we have all the links fixed, but I didn't see that as part of this PR.

File: sdk/eventhubs/azure-messaging-eventhubs/README.md (hunk 1 above)
Comment (sima-zhu) [reply sdk/eventhubs/azure-messaging-eventhubs/README.md:84]: This is where we do match and replace. Will fix it.

File: eng/common/pipelines/templates/steps/verify-links.yml [hunk 2]
```diff
@@ -4,7 +4,7 @@ parameters:
   WorkingDirectory: '$(System.DefaultWorkingDirectory)'
   ScriptDirectory: 'eng/common/scripts'
   Recursive: $false
-  CheckLinkGuidance: $false
+  CheckLinkGuidance: $true
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/verify-links.yml:7]: Lets not modify this eng/common file but instead explicitly pass true where we use it in the java repo.

**Comment by weshaggard** [comment]
Manually triggered java - aggregate-reports jobs https://dev.azure.com/azure-sdk/internal/_build/results?buildId=542853&view=logs&j=ad792ccc-30b2-59a1-5b53-546a79d88828 to verify.

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml [hunk 3]
```diff
@@ -164,6 +157,7 @@ jobs:
       - template: ../../../common/pipelines/templates/steps/verify-links.yml
         parameters:
           Directory: sdk/${{ parameters.ServiceDirectory }}
+          CheckLinkGuidance: $true
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml:160]: I suspect this might also need to be set for the aggregate-reports pipeline.

**Review by weshaggard — APPROVED** [review]
Quick scan looks good. Lets wait for the java - aggregate-reports pipeline so it can validate all the links in the repo

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml (hunk 3 above)
Comment (sima-zhu) [reply eng/pipelines/templates/jobs/archetype-sdk-client.yml:160]: I did not enable aggregate-report yml. Will trigger the aggregate-report again once I check in the changes.

Comment (weshaggard) [reply eng/pipelines/templates/jobs/archetype-sdk-client.yml:160]: I'm Ok with that as long as you quickly follow-up on any broken links as I want to make sure we don't accidently break any pipelines.

**Comment by weshaggard** [comment]
@sima-zhu  I'm going to force merge this right now. Please follow-up with a change the aggregate-reports pipeline to enable the guidance so that we can ensure there are no other breaks.

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml (hunk 3 above)
Comment (sima-zhu) [reply eng/pipelines/templates/jobs/archetype-sdk-client.yml:160]: I plan to work on lang repos subsequently as work altogether will cause a spike unfortunately.

**Pull request merged**